src/synth/fluid_synth.c

```diff
@@ -3897,19 +3895,15 @@ fluid_synth_set_reverb_on(fluid_synth_t* synth, int on)
  * @note Currently private to libfluidsynth.
  */
 int
-fluid_synth_set_reverb_preset(fluid_synth_t* synth, int num)
-{
-  int i = 0;
-  while (revmodel_preset[i].name != NULL) {
-    if (i == num) {
-      fluid_synth_set_reverb (synth, revmodel_preset[i].roomsize,
-                              revmodel_preset[i].damp, revmodel_preset[i].width,
-                              revmodel_preset[i].level);
-      return FLUID_OK;
-    }
-    i++;
-  }
-  return FLUID_FAILED;
+fluid_synth_set_reverb_preset(fluid_synth_t* synth, unsigned int num)
```

Comment:
I would leave `num` signed and check for `<0`. Else you would have to adjust it's usage in `fluid_handle_reverbpreset()`.

Reply:
Done, but if I could say my opinion, this was not a problem because the test is safe even if you pass a negative number at upper level (I had seen the call to atoi() before doing this change, so I did it intentionally).

Reply:
Given 32 bit integers: If the user enters -4294967295 it would select preset 2, which I dont quite consider to be correct ;)

Reply:
32 bit signed integers range from -2147483648 to 2147483647.
2147483647 is 0x7FFFFFFF
-2147483648 is 0x80000000, that is 2147483648 if tested as unsigned.
So, for the properties of the boolean arithmetic, the smallest negative signed integer is always greater than biggest positive signed value. That's why it is safe to do this assumption and why the compilers too sometimes converts things like "if (a<0 || a>max)" to "if ((unsigned )a>max)".

i=atoi("-4294967295") always returns 1, because that value overflows the limits of the type, so it does not matter too much to have the test positive or not... :)

Reply:
Oh yeah, completely confused, sry. Should go to bed.